### DISCUSSION TITLE
Nightly release candidates

### DISCUSSION THREAD

**Discussion by mmachatschek** [keystonejs/keystone#8472] [Feature requests]
In the past we had to wait for some updates (like prisma) for a long time to be released for keystone.

Therefore I suggest that it would be great if keystone had nightly/weekly or any sort of pre releases. Users could test those pre-releases prior to the main release of keystone itself and also benefit from more frequent updates.

My use case would be specifically to have more frequent updates to prisma, that currently tackle a lot of cold start issues for serverless environments.

**Comment by dcousens** [comment]
Hi @mmachatschek,

We actually talked about this recently on the [Keystone 6 public slack](https://app.slack.com/client/T02FLV1HN/C01STDMEW3S/thread/C01STDMEW3S-1680162004.154139) (to signup, go to https://community.keystonejs.com/), but to bring the conversation here:

> A small community question, is anyone here interested in nightly builds for the `@keystone-6/* packages`?
> We are intermittently publishing [to the](https://www.npmjs.com/package/@keystone-6/core?activeTab=versions) `rc` [tag on npm](https://www.npmjs.com/package/@keystone-6/core?activeTab=versions), and with your help we could test and trial new features before they are released.
> You'll need to be participating and reading the pull requests though to keep up with the changes, as we aren't publishing any change summaries for now.

A nightly tag may be a little bit hap-hazard for now, but any help in increasing the stability and reliability of CI/CD pipeline would be appreciated if you're interested.

**Reply by dcousens** [reply]
@mmachatschek probably what we need to resolve is how users can know if a nightly release exists, without polluting our GitHub releases with candidates.  

Reviewing our [`Publish (Snapshot)`](https://github.com/keystonejs/keystone/blob/52b44724a9f0799fc8486cd37c6b8f0408ce7410/.github/workflows/publish_snapshot.yml) workflow, I think we're ready otherwise to put this on a nightly timer...

**Reply by mmachatschek** [reply]
> without polluting our GitHub releases with candidates.

@dcousens they would pollute it in a sense of (there were only some prereleases on the repo)

thanks to changeset there would only be changed packages that would be released to the `rc` tag.

e.g. next.js has a lot of canary releases (some of them multiple times a day), from my point of view this would be acceptable to have some noise of rc tags in the npm releases.

I suppose friday releases (weekly) would be ok too. Over the weekend and the beginning of the next week, users could provide feedback of issues they have with a RC tag. And there would still be the possibility to manually dispatch a release during the week if needed.

**Reply by dcousens** [reply]
@mmachatschek maybe we could setup a scheduled action?
